api/api.go

```diff
@@ -417,6 +421,62 @@ type CloudBackupSchedEnumerateResponse struct {
 	Schedules map[string]CloudBackupScheduleInfo
 }
 
+type Operation string
+
+const (
+	MigrateCluster     = Operation("MigrateCluster")
+	MigrateVolume      = Operation("MigrateVolume")
+	MigrateVolumeGroup = Operation("MigrateVolumeGroup")
+)
+
+type MigrateStatus string
+
+const (
+	MigrateStatusQueued       = MigrateStatus("Queued")
+	MigrateStatusInitializing = MigrateStatus("Initializing")
+	MigrateStatusInProgress   = MigrateStatus("InProgress")
+	MigrateStatusFailed       = MigrateStatus("Failed")
+	MigrateStatusComplete     = MigrateStatus("Complete")
+)
+
+type MigrateStage string
+
+const (
+	MigrateStageBackup       = MigrateStage("Backup")
+	MigrateStageRestore      = MigrateStage("Restore")
+	MigrateStageVolumeUpdate = MigrateStage("VolumeUpdate")
+	MigrateStageDone         = MigrateStage("Done")
+)
+
+type CloudMigrateInfo struct {
+	ClusterId       string
+	LocalVolumeId   string
+	LocalVolumeName string
+	RemoteVolumeId  string
+	CloudSnapId     string
+	CurrentStage    MigrateStage
+	Status          MigrateStatus
+	LastUpdate      time.Time
+	LastSuccess     time.Time
+}
+
+type CloudMigrateStartRequest struct {
+	Operation Operation
+	ClusterId string
+	// Depending on Operation type this can be VolumeID or VolumeGroupID
+	TargetId string
```

Comment:
for migrating volume group, should we allow labels as well ( along with group iD)? Since group snapshot api allows both groupID and labels.